boat_simulator/nodes/low_level_control/controller.py

```diff
@@ -0,0 +1,293 @@
+"""Low level control logic for actuating the rudder and the sail."""
+
+from abc import ABC, abstractmethod
+from math import atan2, cos, pi, sin
+from typing import List
+
+import numpy as np
+
+from boat_simulator.common.types import Scalar
+from boat_simulator.common.utils import bound_to_180
+
+
+class ActuatorController(ABC):
+    """Abstract class for rudder and sail actuation mechanisms.
+
+    Attributes:
+        `current_heading` (Scalar): Current boat heading direction in degrees.
+        `desired_heading` (Scalar): Target boating heading direction in degrees.
+        `current_control_ang` (Scalar): Current control mechanism angle in degrees.
+        `time_step` (Scalar): Time per iteration given in degrees/second.
+        `kp` (Scalar): Proportional constant when calculating error.
+        `cp` (Scalar): Timeseries of error values computed over time.
+        `prev_control` (list): Timeseries of control angles computed over time.
+        `time_series` (list): Timeseries of running time values corresponding to control angles
```

Comment:
Yeah I thought about whether or not those parameters needed to be implemented in the abstract class, they are individual to the sail and rudder, so it would make sense to have them separate for each controller. Prev_control and time_series weren't needed in the implementation, I just thought it would be helpful to have them for graphing and visualizing in the future possibly but they aren't fundamental to the implementation at all.